glean-core/python/glean/glean.py

```diff
@@ -315,32 +297,35 @@ def set_upload_enabled(cls, enabled: bool) -> None:
         Args:
             enabled (bool): When True, enable metric collection.
         """
-        if cls.is_initialized():
+        # Changing upload enabled always happens asynchronous.
+        # That way it follows what a user expect when calling it inbetween other calls:
+        # It executes in the right order.
+        #
+        # Because the dispatch queue is halted until Glean is fully initialized
+        # we can safely enqueue here and it will execute after initialization.
+        @Dispatcher.launch
```

Comment:
This is a nit, but since there is literally nothing in this method that needs to happen on the main thread, you could just put the `@Dispatcher.task` decorator on the method itself and not have a nested function.

Reply:
Thanks, that helps improve my python foo